cardano-cli/src/Cardano/CLI/Conway/Commands.hs

```diff
@@ -35,16 +43,40 @@ runGovernanceCreateVoteCmd
   -> VoteChoice
   -> VoterType
   -> TxIn
-  -> StakeIdentifier
+  -> VerificationKeyOrFile StakePoolKey
   -> ConwayVoteFile Out
   -> ExceptT GovernanceCmdError IO ()
-runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn votingStakeCred oFp = do
-  stakeCred <- firstExceptT StakeCredGovCmdError $ getStakeCredentialFromIdentifier votingStakeCred
-  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe stakeCred
+runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn stakePoolVerKeyOrFile oFp = do
+  stakePoolVerKey <- firstExceptT VotingCredentialDecodeError
+    . newExceptT
+    $ readVerificationKeyOrFile AsStakePoolKey stakePoolVerKeyOrFile
+
+  let stakePoolVerificationKeyHash = unStakePoolKeyHash $ verificationKeyHash stakePoolVerKey
+  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ eraDecodeVotingCredential sbe $ Plain.serialize $ Ledger.KeyHashObj stakePoolVerificationKeyHash
+
   let govActIdentifier = makeGoveranceActionIdentifier sbe govActionTxIn
       voteProcedure = createVotingProcedure sbe vChoice cType govActIdentifier votingCred
   firstExceptT WriteFileError . newExceptT $ obtainEraPParamsConstraint sbe $ writeFileTextEnvelope oFp Nothing voteProcedure
+  where
+    obtainCryptoConstraints
+      :: ShelleyBasedEra era
+      -> ((Crypto (EraCrypto (ShelleyLedgerEra era))) => a)
+      -> a
+    obtainCryptoConstraints ShelleyBasedEraShelley f = f
+    obtainCryptoConstraints ShelleyBasedEraAllegra f = f
+    obtainCryptoConstraints ShelleyBasedEraMary    f = f
+    obtainCryptoConstraints ShelleyBasedEraAlonzo  f = f
+    obtainCryptoConstraints ShelleyBasedEraBabbage f = f
+    obtainCryptoConstraints ShelleyBasedEraConway  f = f
 
+    eraDecodeVotingCredential
+      :: ShelleyBasedEra era
+      -> BL.ByteString
+      -> Either Plain.DecoderError (VotingCredential era)
+    eraDecodeVotingCredential sbe' bs = obtainCryptoConstraints sbe' $
+      case Plain.decodeFull bs of
+        Left e -> Left e
+        Right x -> Right $ VotingCredential x
 
 
 runGovernanceNewConstitutionCmd
```

Comment:
Shoud this also be changed to include SPO key?

Reply:
Yeah for now that's ok but leave a comment similar to the one above I suggested.